File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) {
        string ans = "";

        for (int i = 0; i < word.size(); i++) {
            if (ans.size() > 0) {
                ans += word[i];
            }
            else if (word[i] == ch) {
                for (int j = 0; j < i + 1; j++) {
                    ans = word[j] + ans;
                }
            }
        }
        return ans.size() > 0 ? ans : word;
    }
};

File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.go
func reversePrefix(word string, ch byte) string {
    var ans string

    for i := 0; i < len(word); i++ {
        if (word[i] == ch) {
            ans = word[i + 1: ]
            for j := 0; j < i + 1; j++ {
                ans = string(word[j]) + ans
            }
            return ans
        }
    }
    return word
}